packages/@aws-cdk/aws-ec2/lib/network-acl.ts

```diff
@@ -0,0 +1,411 @@
+   */
+  readonly networkAclEntryName?: string;
+ /**
+  * The IPv4 CIDR range to allow or deny, in CIDR notation (for example, 172.16.0.0/24).
+  * Requirement is conditional: You must specify the CidrBlock or Ipv6CidrBlock property.
+  */
+  readonly cidrBlock: string;
+
+  /**
+   * Whether this rule applies to egress traffic from the subnet (true) or ingress traffic to the subnet (false).
+   * By default, AWS CloudFormation specifies false.
+   * @default false
+   */
+  readonly egress: boolean;
+
+  /**
+   * The Internet Control Message Protocol (ICMP) code and type.
+   * Requirement is conditional: Required if specifying 1 (ICMP) for the protocol parameter.
+   *
+   * @default undefined
+   */
+  readonly icmp?: Icmp
+
+  /**
+   * The IPv6 network range to allow or deny, in CIDR notation.
+   * Requirement is conditional: You must specify the CidrBlock or Ipv6CidrBlock property.
+   *
+   * @default undefined
+   */
+  readonly ipv6CidrBlock?: string
+
+  /**
+   * The ID of the network ACL.
+   */
+  readonly networkAcl: INetworkACL
+
+  /**
+   * The range of port numbers for the UDP/TCP protocol.
+   * Conditional required if specifying 6 (TCP) or 17 (UDP) for the protocol parameter.
+   *
+   * @default undefined
+   */
+  readonly portRange?: PortRange
+
+  /**
+   * The protocol number. A value of "-1" means all protocols. If you specify "-1"
+   * or a protocol number other than "6" (TCP), "17" (UDP), or "1" (ICMP), traffic
+   * on all ports is allowed, regardless of any ports or ICMP types or codes that
+   * you specify. If you specify protocol "58" (ICMPv6) and specify an IPv4 CIDR
+   * block, traffic for all ICMP types and codes allowed, regardless of any that
+   * you specify. If you specify protocol "58" (ICMPv6) and specify an IPv6 CIDR
+   * block, you must specify an ICMP type and code.
+   *
+   * @default 17
+   */
+  readonly protocol: number;
+
+  /**
+   * Whether to allow or deny traffic that matches the rule; valid values are "allow" or "deny".
+   */
+  readonly ruleAction: string
+
+  /**
+   * Rule number to assign to the entry, such as 100. ACL entries are processed in ascending order by rule number.
+   * Entries can't use the same rule number unless one is an egress rule and the other is an ingress rule.
+   */
+  readonly ruleNumber: number
+}
+
+export class NetworkAclEntry extends NetworkAclEntryBase {
+  public readonly  cidrBlock: string;
+  public readonly  egress: boolean;
+  public readonly  icmp?: Icmp;
+  public readonly  ipv6CidrBlock?: string;
+  public readonly  networkAcl: INetworkACL;
+  public readonly  portRange?: PortRange;
+  public readonly  protocol: number;
+  public readonly  ruleAction: string;
+  public readonly  ruleNumber: number;
+
+  constructor(scope: Construct, id: string, props: NetworkAclEntryProps) {
+    super(scope, id, {
+      physicalName: props.networkAclEntryName
+    });
+
+    this.cidrBlock = props.cidrBlock;
```

Comment:
Not sure all these assignments to members are necessary. Where are their values used?

Reply:
I was assuming someone my wish read back the values , makes sense to remove if never used.
Happy to remove.

Reply:
all removed